library/test/src/lib.rs

```diff
@@ -182,8 +182,8 @@ fn make_owned_test(test: &&TestDescAndFn) -> TestDescAndFn {
 /// and checks for a `0` result.
 pub fn assert_test_result<T: Termination>(result: T) {
     let code = result.report().to_i32();
-    assert_eq!(
-        code, 0,
+    assert!(
```

Comment:
`assert_eq` also accepts an error message

Reply:
It was assert_eq with error message. It results in confusing output, please see examples in initial comment or diff's comment

Reply:
The `assert_eq` here doesn't help, because the message already reports the [`code`](https://doc.rust-lang.org/std/process/struct.ExitCode.html) value, if it needs to be reported at all. Instead, it harms as shown in the description, in #69517 and in #59263.

In fact the fix was already approved as #59263 but it landed in an outdated repository.